packages/buefy-next/src/components/upload/Upload.vue

```diff
@@ -71,44 +80,35 @@ export default {
         }
     },
     emits: ['invalid', 'update:modelValue'],
-    data() {
+    data(): BUploadData {
         return {
-            newValue: this.modelValue,
+            newValue: this.modelValue ?? null,
             dragDropFocus: false,
             _elementRef: 'input'
         }
     },
     computed: {
         disabledOrUndefined() {
-            // On Vue 3, setting a boolean attribute `false` does not remove it,
-            // `true` or `undefined` has to be given to remove it.
             return this.disabled || undefined
         }
     },
     watch: {
-        /**
-         *   When v-model is changed:
-         *   1. Set internal value.
-         *   2. Reset internal input file value
-         *   3. If it's invalid, validate again.
-         */
         modelValue(value) {
             this.newValue = value
             if (!value || (Array.isArray(value) && value.length === 0)) {
-                this.$refs.input.value = null
+                (this.$refs.input as HTMLInputElement).value = ""
             }
             !this.isValid && !this.dragDrop && this.checkHtml5Validity()
         }
     },
     methods: {
-        /**
-        * Listen change event on input type 'file',
-        * emit 'input' event and validate
-        */
-        onFileChange(event) {
+        onFileChange(event: Event) {
             if (this.disabled || this.loading) return
             if (this.dragDrop) this.updateDragDropFocus(false)
-            const value = event.target.files || event.dataTransfer.files
+            const value = (event.target as HTMLInputElement)?.files
+                            ?? (event as DragEvent).dataTransfer?.files
+                            ?? [];
```

Comment:
As a conservative guy I would do something like the following so that the original behavior is retained:
```ts
const value = (event.target as HTMLInputElement)!.file || (event as DragEvent).dataTransfer!.files
if (value!.length === 0) {
```
Here the non-null assertion operator (`!`) is used. It crashes if the value is actually `null` or `undefined`, but the behavior is consistent with the original implementation.